src/pybind/mgr/dashboard/CMakeLists.txt

```diff
@@ -28,33 +28,39 @@ endfunction(add_npm_command)
 
 if(WITH_MGR_DASHBOARD_FRONTEND AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64|arm|ARM")
 
-set(mgr-dashboard-nodeenv ${CMAKE_CURRENT_BINARY_DIR}/node-env)
+if(NOT WITH_SYSTEM_NPM)
+  set(mgr-dashboard-node-env ${CMAKE_CURRENT_BINARY_DIR}/node-env)
 
-add_custom_command(
-  OUTPUT "${mgr-dashboard-nodeenv}/bin/npm"
-  COMMAND ${CMAKE_SOURCE_DIR}/src/tools/setup-virtualenv.sh --python=${MGR_PYTHON_EXECUTABLE} ${mgr-dashboard-nodeenv}
-  COMMAND ${mgr-dashboard-nodeenv}/bin/pip install nodeenv
-  COMMAND ${mgr-dashboard-nodeenv}/bin/nodeenv -p -n 8.11.3
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  COMMENT "dashboard nodeenv is being installed"
-)
-add_custom_target(mgr-dashboard-nodeenv
-  DEPENDS ${mgr-dashboard-nodeenv}/bin/npm
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-)
+  add_custom_command(
+    OUTPUT "${mgr-dashboard-node-env}/bin/npm"
+    COMMAND ${CMAKE_SOURCE_DIR}/src/tools/setup-virtualenv.sh --python=${MGR_PYTHON_EXECUTABLE} ${mgr-dashboard-node-env}
+    COMMAND ${mgr-dashboard-node-env}/bin/pip install nodeenv
+    COMMAND ${mgr-dashboard-node-env}/bin/nodeenv -p -n 8.11.3
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "dashboard nodeenv is being installed"
+  )
+  add_custom_target(mgr-dashboard-nodeenv
+    DEPENDS ${mgr-dashboard-node-env}/bin/npm
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  )
+  set(mgr-dashboard-nodeenv mgr-dashboard-nodeenv)
+  set(nodeenv NODEENV)
+else()
+  set(mgr-dashboard-nodeenv )
+  set(nodeenv "")
+endif()
 
+add_custom_target(mgr-dashboard-frontend-deps
```

Comment:
why shall we move `add_custom_target(mgr-dashboard-frontend-deps ...)` up here?

Reply:
you are moving `add_custom_target(mgr-dashboard-frontend-deps ...)` around in this commit. i am not sure why you are doing so.

Reply:
@tchaikov 
Will chcek, wasn't aware of it. But also had already to many versions on my desk.